dockerfiles/Dockerfile

```diff
@@ -76,6 +76,66 @@ WORKDIR /srv/docsrs
 # Tini is a small init binary to properly handle signals
 CMD ["/usr/bin/tini", "/usr/local/bin/cratesfyi", "start-web-server", "0.0.0.0:80"]
 
+########################
+#  Build server stage  #
+########################
+
+FROM ubuntu:22.04 AS build-server
+
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get upgrade -y \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y \
+        ca-certificates \
+        tini \
+        docker.io \
+        build-essential \
+        gcc \
+        pkg-config \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY --from=build /build/target/release/cratesfyi /usr/local/bin
+
+WORKDIR /srv/docsrs
+# Tini is a small init binary to properly handle signals
+CMD ["/usr/bin/tini", "/usr/local/bin/cratesfyi", "start-build-server"]
+
+############################
+#  Registry watcher stage  #
+############################
+
+FROM ubuntu:22.04 AS registry-watcher
+
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get upgrade -y \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y \
+        ca-certificates \
+        tini \
+        git \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY --from=build /build/target/release/cratesfyi /usr/local/bin
+
+WORKDIR /srv/docsrs
+# Tini is a small init binary to properly handle signals
+CMD ["/usr/bin/tini", "/usr/local/bin/cratesfyi", "start-registry-watcher"]
+
+###############
+#  CLI stage  #
+###############
+
+FROM ubuntu:22.04 AS cli
+
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get upgrade -y \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y \
+        ca-certificates \
+        tini \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY --from=build /build/target/release/cratesfyi /usr/local/bin
+
+ENTRYPOINT ["/usr/local/bin/cratesfyi"]
+
 ##################
 #  Output stage  #
 ##################
```

Comment:
after this PR, what are we using the output stage for? 

If for nothing, we could also remove `entrypoint.sh`, which is only used there?

Reply:
Nothing that I can think of, I'll clean it up too.